applications/ShallowWaterApplication/python_scripts/eulerian_primitive_var_solver.py

```diff
@@ -1,23 +1,43 @@
 from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
 # importing the Kratos Library
 import KratosMultiphysics
-import KratosMultiphysics.ShallowWaterApplication as KratosShallow
+import KratosMultiphysics.ShallowWaterApplication as Shallow
 
 # Check that KratosMultiphysics was imported in the main script
 KratosMultiphysics.CheckForPreviousImport()
 
 ## Import base class file
-import shallow_water_base_solver
+from shallow_water_base_solver import ShallowWaterBaseSolver
 
-def CreateSolver(model_part, custom_settings):
-    return EulerianPrimitiveVarSolver(model_part, custom_settings)
+def CreateSolver(model, custom_settings):
+    return EulerianPrimitiveVarSolver(model, custom_settings)
+
+class EulerianPrimitiveVarSolver(ShallowWaterBaseSolver):
+    def __init__(self, model, custom_settings):
+        super(EulerianPrimitiveVarSolver, self).__init__(model, custom_settings)
+
+        # Set the element and condition names for the replace settings
+        self.element_name = "EulerPrimVarElement"
+        self.condition_name = "Condition"
+        self.min_buffer_size = 2
 
-class EulerianPrimitiveVarSolver(shallow_water_base_solver.ShallowWaterBaseSolver):
     def AddDofs(self):
-        super(EulerianPrimitiveVarSolver,self)._AddPrimitiveDofs()
+        KratosMultiphysics.VariableUtils().AddDof(KratosMultiphysics.VELOCITY_X, self.main_model_part)
+        KratosMultiphysics.VariableUtils().AddDof(KratosMultiphysics.VELOCITY_Y, self.main_model_part)
+        KratosMultiphysics.VariableUtils().AddDof(Shallow.HEIGHT, self.main_model_part)
+
+        if self._IsPrintingRank():
+            self.print_on_rank_zero("::[EulerianPrimitiveVarSolver]::", "Shallow water solver DOFs added correctly.")
```

Comment:
This is "double defined", if you ask for the printing rank, them you can use the logger directly